compose/mpp/demo/src/commonMain/kotlin/androidx/compose/mpp/demo/FontRasterization.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.mpp.demo
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.text.BasicText
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.ExperimentalTextApi
+import androidx.compose.ui.text.FontHinting
+import androidx.compose.ui.text.FontRasterizationSettings
+import androidx.compose.ui.text.FontSmoothing
+import androidx.compose.ui.text.PlatformParagraphStyle
+import androidx.compose.ui.text.PlatformTextStyle
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.unit.dp
+
+@OptIn(ExperimentalTextApi::class)
+@Composable
+fun FontRasterization() {
+    MaterialTheme {
+        val state = rememberScrollState()
+        Column(Modifier.fillMaxSize().padding(10.dp).verticalScroll(state)) {
+            val hintingOptions = listOf(FontHinting.None, FontHinting.Slight, FontHinting.Normal, FontHinting.Full)
+            val isAutoHintingForcedOptions = listOf(false, true)
+            val subpixelOptions = listOf(false, true)
+            val smoothingOptions = listOf(FontSmoothing.None, FontSmoothing.AntiAlias, FontSmoothing.SubpixelAntiAlias)
```

Comment:
We map Java [`RenderingHints`](https://docs.oracle.com/javase/8/docs/api/java/awt/RenderingHints.html#VALUE_TEXT_ANTIALIAS_LCD_HRGB) to the `PixelGeometry` they represent and pass that to the `SkiaLayer`

Reply:
@igordmn good point! I totally missed this part about layer geometry. As Manuel, said before we're relying on AWT here, and it works fine in most cases, the only problem is that our current approach isn't reactive, AWT doesn't allow request this settings per screen, or to be notified when it changes. So, it's probably a good idea to investigate a possibility to implement such API along with retrieving OS defaults.

Modern OLED screens are usually high-dpi, so with `PixelGeometry.UNKNOWN` Skia just disable subpixel anti-aliasing for them (not only for fonts but e.g. for line drawing), but it still looks good.

So I would merge this PR first in current state, (`FontSmoothing.SubpixelAntiAlias` doesn't make any harm because it's ignored with `PixelGeometry.UNKNOWN`), and proceed with this work later.

Reply:
Or in case if you have a strong opinion, we could tune platform defaults in Compose and override it in Fleet

Reply:
> so with PixelGeometry.UNKNOWN Skia just disable subpixel anti-aliasing for

See the screenshot, it differs. I am not sure if it is also a valid antialiasing or it is some of strange form of subpixel antialising, or it is broken.

Reply:
Ok, I see now, can we agree on disabling `FontSmoothing.SubpixelAntiAlias` in Compose `PlatformDefault` and merge it?

Reply:
> AWT doesn't allow request this settings per screen

If AWT relies on global system settings, which is also applied to all applications, maybe it is fine, and we don't have to

> be notified when it changes

On desktop orientation changes are very rare, and it seems like a minor feature.

So, we can can just read RenderingHints to determine the current geometry and the system settings?

One thing we still need to decide - should we change only `SkiaLayer(geometry=`, or also the default `smoothing` property.

Reply:
> Ok, I see now, can we agree on disabling FontSmoothing.SubpixelAntiAlias in Compose PlatformDefault and merge it?

Sure, in this state we can merge it

Reply:
> On desktop orientation changes are very rare, and it seems like a minor feature.

Just to be clear here: This is not about orientation changes but about dragging the window from a screen with one pixel geometry to another screen with a different one. This is relevant if one of the screens is rotated 90 degrees or if one of them has RGB and the other one BGR, or similar cases.

As a reference: This is a feature that Windows natively also doesn't get right yet, and many applications will do crappy text rendering if your displays don't have the same geometry. It doesn't mean that we should be crappy at that, too. But I think it will not surprise users when they see that this doesn't work really well.

Reply:
>  One thing we still need to decide - should we set only SkiaLayer(geometry=, or also the default smoothing property.

afaik, we need both of them, `geometry` affects drawing in general, and if you want font to be smoothed we need to configure its property  too

Reply:
Yes, sure.

The question was - can we just keep "SubpixelAntialiasing" and only change `geometry` to UKNOWN if it is disabled in the system.

My intuition tells that we can't, as `geometry` isn't subpixel related by its spec. It can be used by something else as well.